src/schema/rules/sidecars/events.yaml

```
@@ -12,3 +12,14 @@ StimulusPresentation:
   fields:
     StimulusPresentation: recommended
     VisionCorrection: optional
+
+StimulusPresentation_for_eyetracking:
+  selectors:
+    - suffix == "events"
+    - associations.physio.sidecar.PhysioType == "eyetrack"
+  fields:
+    StimulusPresentation.ScreenDistance: required
+    StimulusPresentation.ScreenOrigin: required
+    StimulusPresentation.ScreenRefreshRate: recommended
+    StimulusPresentation.ScreenResolution: required
+    StimulusPresentation.ScreenSize: required
```

looking above at https://github.com/bids-standard/bids-specification/pull/1128/files#diff-9c80bef9a8601b9410f41180bfb2631eee51fb67e7f5727ca0be4e4dfc99d844R3856 , shouldn't this be

```suggestion
    StimulusPresentation.ScreenSize: recommended
```

if that is what possibly triggering current test fails like

```
FAILED src/bidsschematools/tests/test_schema.py::test_add_legal_field - jsonschema.exceptions.ValidationError: {'ScreenSize': 'required'} is not valid under any of the given schemas
```
?

---

I have the filing of an issue to the validator pending. I think this is caused by the new "association" that mutates the value of some entries in StimulusPresentation when eyetracking is present.